### ISSUE TITLE
UserWarning: Detected call of `lr_scheduler.step()` before `optimizer.step()`

### ISSUE THREAD

**Issue by RahulSinghalChicago** [sithu31296/semantic-segmentation#28]
/usr/local/lib/python3.7/dist-packages/torch/optim/lr_scheduler.py:134: UserWarning: Detected call of `lr_scheduler.step()` before `optimizer.step()`. In PyTorch 1.1.0 and later, you should call them in the opposite order: `optimizer.step()` before `lr_scheduler.step()`.  Failure to do this will result in PyTorch skipping the first value of the learning rate schedule. See more details at https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate
  "https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate", UserWarning)

**Comment by sithu31296** [comment]
I see you have enabled the mixed precision. Pytorch still has an error in the first iteration if you use mixed precision training. You can trun off AMP or you can still train even with that error.

**Comment by RahulSinghalChicago** [comment]
Yes, it continues to train with the Warning. thank you!
